AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt

```
@@ -642,11 +645,24 @@ open class AnkiActivity : AppCompatActivity, SimpleMessageDialogListener, Collec
             activitySuperOnCreate = { state -> super.onCreate(state) }
         )
 
+    /**
+     * Schedules a job to save collection in background and returns immediately
+     */
     fun saveCollectionInBackground(syncIgnoresDatabaseModification: Boolean = false) {
         if (CollectionHelper.instance.colIsOpen()) {
-            launchCatchingTask {
+            lifecycleScope.launch {
                 Timber.d("saveCollectionInBackground: start")
-                CollectionManager.withCol { saveCollection(this, syncIgnoresDatabaseModification) }
+                withCol {
+                    try {
+                        if (syncIgnoresDatabaseModification) {
+                            SyncStatus.ignoreDatabaseModification { col.save() }
+                        } else {
+                            col.save()
+                        }
+                    } catch (e: RuntimeException) {
+                        Timber.e(e, "Error on saving deck in background")
+                    }
```

I want a second opinion here. We are only catching RuntimeException here, so if any other exception is thrown it would crash the app in background (cuz this fun is called from onStop()). When user then returns to the app, it would simply show App Crashed dialog (typical 'App Keeps Crashing' message) and user won't have any idea what happened. So either I can keep launchCathingTask and remove the try-catch which would catch and show a dialog when user returns to the app or can silently log "Exception" instead of "RuntimeException" to logcat.

---

Which would be more preferred?

---

Honestly, I’ve no idea.
Main trouble is that I can’t even guess what we are trying to catch here. What can actually occurs. 
So ideally, what I’d want is to go through git history (I suggest using `gitk` to do so, but any tools you know how to use is fine), figure out when this was added for the first time (looking at the first introduction of "Error on saving deck in background" should suffices), and read the comment to understand what we were trying to do.

If we don’t know what was going on, I’m fine with catching exactly the same thing as before and leaving your very relevant question as a separate issue.
I believe actually the issue can be generalized. We can just ask to go over each general catch (one where we catch all Exceptions, RuntimeException or even Throwable) and asses them individually. 
For this one, I believe the right things to do is to throw an Exception to Acra, so that we can learn what bugs are actually encounted, and, hopefully, correct them, or if it never occurs, remove the try/catch.
Obviously, this is blocked by actually creating releases, so it must just remains an issue. 
At best, put a todo with a link to the issue

---

Makes sense to catch `Exception` and send it to ACRA.

---

Like @ShridharGoel  I would catch `Exception` there. Sending to ACRA will trigger the same issue of a dialog appearing from the background(all this time the app functioned without sending a report).

So I would say catch `Exception` and add a todo in the catch block:
```

} catch (e: Exception) {
    Timber.e(e, "Error on saving deck in background")
    // TODO should this error be reported through our error reporting service?
}
```

---

done